frontend/src/components/widgets/Form/FormSubmitButton.test.tsx

```diff
@@ -0,0 +1,94 @@
+/**
+ * @license
+ * Copyright 2018-2021 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react"
+import { shallow } from "lib/test_util"
+import { WidgetStateManager } from "lib/WidgetStateManager"
+
+import UIButton, { Kind } from "components/shared/Button"
+
+import { Button as ButtonProto } from "autogen/proto"
+import FormSubmitButton, { Props } from "./FormSubmitButton"
+
+jest.mock("lib/WidgetStateManager")
+
+const sendBackMsg = jest.fn()
+
+const getProps = (buttonProps: Partial<Props> = {}): Props => ({
+  element: ButtonProto.create({
+    id: "1",
+    label: "Label",
+    formId: "test_id",
+  }),
+  width: 0,
+  disabled: false,
+  hasPendingChanges: false,
+  // @ts-ignore
+  widgetMgr: new WidgetStateManager(sendBackMsg),
+  ...buttonProps,
+})
+
+describe("FormSubmitButton", () => {
+  const props = getProps()
+  const wrapper = shallow(<FormSubmitButton {...props} />)
+  const wrappedUIButton = wrapper.find(UIButton)
+
+  it("renders without crashing", () => {
+    expect(wrapper).toBeDefined()
+  })
+
+  it("should have correct className and style", () => {
+    const wrappedDiv = wrapper.find("div").first()
+    const { className, style } = wrappedDiv.props()
+    const splittedClassName = className?.split(" ")
```

Comment:
Nit: "splitted" isn't a word; how about "classNameParts" instead?

Reply:
Lol. I didn't pay attention when I copied it. It turns out we have a bunch of `splittedClassName`s across our codebase. I'll create a separate PR to clean them up.